### PR TITLE
Enable optional tensorList fallback to cpu.

### PR DESCRIPTION
**Pull request by Shan19900305**
add optional tensorList fallback to cpu.
Add testcases and old pr is: https://github.com/pytorch/pytorch/pull/106449

@bdhirsh

### CODE REVIEW

File: aten/src/ATen/native/CPUFallback.cpp [hunk 1]
```diff
@@ -108,6 +108,22 @@ void cpu_fallback(const c10::OperatorHandle& op, torch::jit::Stack* stack, bool
...
+        if (!opt_tensors[i].has_value() || !opt_tensors[i]->defined()) continue;
+        need_convert_tensors.push_back(opt_tensors[i].value());
+        need_convert_tensors_index.push_back(i);
+      }
+      auto cpu_tensors = to_cpu(need_convert_tensors);
+      for (const auto i : c10::irange(need_convert_tensors_index.size())) {
+        auto idx = need_convert_tensors_index[i];
+        opt_tensors[idx] = cpu_tensors[i];
+      }
+      (*stack)[arguments_begin + idx] = c10::IValue(opt_tensors);
```
Comment (Skylion007) [inline aten/src/ATen/native/CPUFallback.cpp:126]:
```suggestion
      (*stack)[arguments_begin + idx] = c10::IValue(std::move(opt_tensors));
```
Recently added move ctor.

**Review by bdhirsh — APPROVED** [review]
thanks for the test!

**Comment by Shan19900305** [comment]
@bdhirsh @Skylion007 I got failed even without this pr. 

`
  File "/projs/platform/shangang/anaconda3/envs/shangang_conda/lib/python3.9/site-packages/torch/_export/__init__.py", line 390, in aot_compile
    constraints = _process_dynamic_shapes(f, args, kwargs, dynamic_shapes)
  File "/projs/platform/shangang/anaconda3/envs/shangang_conda/lib/python3.9/site-packages/torch/export/dynamic_shapes.py", line 470, in _process_dynamic_shapes
    for tensor, shape in tree_zip(combined_args, dynamic_shapes):
  File "/projs/platform/shangang/anaconda3/envs/shangang_conda/lib/python3.9/site-packages/torch/export/dynamic_shapes.py", line 355, in tree_zip
    raise UserError(
torch._dynamo.exc.UserError: Expected ({0: <class '__main__.dim0_x'>}, None, None) to have 1 items

To execute this test, run the following from the base repo dir:
     python test/inductor/test_memory_planning.py -k test_abi_compatible

`

**Pull request closed without merging**